### PR TITLE
[ADD] website_portal_sale: Backport from Odoo V9.

### PR DESCRIPTION
**Pull request by sergio-teruel**
This PR replace #69 
CC @LeartS @rafaelbn

### CODE REVIEW

**Comment by oscarolar** [comment]
This PR should be in e-commerce I think it must be replaced with this https://github.com/OCA/e-commerce/pull/77

**Comment by pedrobaeza** [comment]
Indeed it should be in e-commerce. It neverminds if it's done by one or the other, but please respect author history.

**Comment by LeartS** [comment]
I don't understand why it should be on e-commerce. It doesn't depends on
website_sale and it can work and have a purpose without the e-commerce
installed.

Regarding closing this PR, no problems. Or you can make PR against my
branch and I'll merge.

**Comment by oscarolar** [comment]
@LeartS You are right, there is no need of website_sale so this is Ok to be here, this is just a website feature, I made the PR with the fixes @sergio-incaser provided. https://github.com/LeartS/website/pull/2

**Comment by rafaelbn** [comment]
- This PR doesn't need website_sale for work, it means must be here in website. So this is the correct repo.
- @sergio-incaser must respect author history and he did.
- https://github.com/OCA/website/pull/69 PR was validated by @pedrobaeza here: https://github.com/OCA/website/pull/69#issuecomment-151551161 so technically it's OK
- Let's test it

**Comment by rafaelbn** [comment]
```
+Go to Sales > Customer > More options > Portal Access Management and invite it
+to access at portal.
+
+When a customer clicked to his name > my account then he can see his orders if
+hi is follower of them.
```

Tested:
- If you go to Sales > Customer, select "Agrolait" > More options > Portal access management this invites to the contacts of the company, and not the company. In this case to "Michel Fletcher" and "Thomas Passot" (creates 2 res.user asociated with this res.parners).
  - Agrolait has 2 sales and 2 invoices
  - Michel has 0 sales and 0 invoices
  - Thomas has 0 sales and 0 invoices
- After this you can login as Michel or Thomas, if you login as Michel you go to the standard portal of Odoo 8 and not to "My account" (http://3119645-111-ab96bd.runbot1.odoo-community.org/my/home)

**[BUG-1]** After login Odoo must redirect the user not to portal, must redirecto to "My account" in the new website_portal_sale

**[BUG-2]** The "My account" of Michel Fletcher has 0 quotes, 0 orders 0 invoices and paymentes. This module must invite "Agrolait" and not to the contacts.

**[BUG-3]** This module shows only the quotations and invoices created after installing the module and not all invoices of the res.partner. It doesn't shows the invoices created before installing it.

**[BUG-4]**
- If you get Agrolait as example, Agrolait has 2 contacts.
- Buy in the ecommers an Ipad with user Michel Flecher
- After buying if you login with Michel Flecher you will have a quotation ->OK
- As admin, Confirm sale order and create invoice, validate invoice
- If you you login with Michel Flecher you will have a quotation and an invocie -> OK
- (trick to give acces to company Agrolait because [BUG-2]) As admin, go to setting - users - create a user with the same email of the company Agrolait, take off all permisions an set only permisions of Portal. After this go to Sales - Customer and Merge the to res.partner Agrolait with same e-mail.
- Log in as Agrolait, the company. Because [BUG-1] you can check that this user Agrolait can see in the "old" v8 portal the quotatin and invoices made by Michel Flecher, this is correct as "Accounting-related settings are managed on the parent company" but (this is the [BUG-4] if you go to the new website_portal_sale, you cannot see any quotation or invoice. To fix: partner company must see all quotations and invoiced of its contacts as "old" v8 portal make.

**Comment by sergio-teruel** [comment]
**[BUG1]** 
This module depends of website_portal, and website_portal after login redirect to home page if the user is not employed. I am not sure if is ncesary to do this.
**[BUG2]** 
This bug is relative to portal module.
**[BUG3]**
I have tested in local instance, and it seem good.
**[BUG4]**
Ok.. I will respect the old portal rules

**Comment by sergio-teruel** [comment]
@rafaelbn Changes done
@pedrobaeza I think that i respect author history.

**Comment by pedrobaeza** [comment]
@sergio-incaser, yeah, of course you did. The sentence was for @oscarolar in his attempt in e-commerce. Sorry for the confusion.

**Comment by sergio-teruel** [comment]
@pedrobaeza Can you force rebuild runbot in this PR?. Thanks
cc @rafaelbn

**Comment by pedrobaeza** [comment]
Why do you need it? It's already built and working.

**Comment by oscarolar** [comment]
**[BUG1]**
@sergio-incaser +1 I dont think this is necessary too.
**[BUG3]**
Tested in local and runbot with an empty db seems to be working ok

**Comment by sergio-teruel** [comment]
@pedrobaeza sorry this PR not contains lastest website_portal so not redirect to home page after login. Runbot is ok.
cc @rafaelbn

**Comment by sergio-teruel** [comment]
@rafaelbn Ready for review

**Comment by oscarolar** [comment]
LGTM :+1:

File: website_portal_sale/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
```
Comment (rafaelbn) [inline website_portal_sale/__init__.py:1]: Must include Monk Software and Odoo, isn't it?

File: website_portal_sale/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
```
Comment (rafaelbn) [inline website_portal_sale/__openerp__.py:1]: Must include Monk Software and Odoo, isn't it?

File: website_portal_sale/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        'Add your sales document in the frontend portal (sales order'
+        ', quotations, invoices)'
+    ),
+    'version': '8.0.1.0.0',
+    'author': 'Odoo SA, '
+              'MONK Software, '
+              'Antiun Ingenieria SA, ',
+    'website': 'https://www.odoo.com/, '
+               'http://www.wearemonk.com, '
+               'http://www.antiuningenieria.com',
```
Comment (rafaelbn) [inline website_portal_sale/__openerp__.py]: s/antiuningenieria/antiun -> www.antiun.com

File: website_portal_sale/controllers/__init__.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
```
Comment (rafaelbn) [inline website_portal_sale/controllers/__init__.py]: Must include Monk Software and Odoo, isn't it?

**Comment by LeartS** [comment]
IMHO it's useless and tedious to put a copyright string as a comment at the top of every .py file and we could just avoid them instead of updating them every time someone makes a change, also because I'm pretty sure everybody puts them there without any real legal knowledge of what it's supposed to do (AFAIK, absolutely nothing). And if we decide they are useful and required, why only on .py files? What about xml, js, css?

This was already discussed here: https://lists.launchpad.net/openerp-community/msg07097.html but I don't think an agreement has been reached.

**Comment by rafaelbn** [comment]
@LeartS I agree

**Comment by rafaelbn** [comment]
Ok, tested. It's OK and improved :+1:

**Comment by rafaelbn** [comment]
There is IMHO a **[BUG2]** uptream, I will inform Odoo

File: website_portal_sale/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    'depends': [
+        'sale',
+        'portal',
+        'website_portal',
+    ],
+    'data': [
+        'templates/website_portal_sale.xml',
+        'templates/website_portal.xml',
+        'templates/website.xml',
+        'demo/sale_order.xml',
```
Comment (LeartS) [inline website_portal_sale/__openerp__.py]: This file is already in the `demo` key

**Comment by LeartS** [comment]
Tested on runbot, I logged in as portal - portal, redirect to portal (but this is expected I seem to understand, right?).
I manually go to the website portal and I see a quotation and an order.
- I click on the quotation, I see it correctly.
- I click on the order, I get a 403 forbidden

**Comment by rafaelbn** [comment]
Where is runbot? I only can see travis y coverage

**Comment by LeartS** [comment]
https://runbot.odoo-community.org/runbot/repo/github-com-oca-website-186

**Comment by rafaelbn** [comment]
Hi @LeartS 

```
    I click on the quotation, I see it correctly.
    I click on the order, I get a 403 forbidden
```

It something related with SO, if you check know you can see order SO008 (SO009 you get forbidden but is SO related). I have created a new quote SO011, and I see quote, order and invoice.

> but this is expected I seem to understand, right

Is expected

**Comment by sergio-teruel** [comment]
@LeartS @oscarolar @rafaelbn Changes done, this module depends of website_portal.
Error is due to website_portal changes not approved yet

**Comment by rafaelbn** [comment]
> Error is due to website_portal changes not approved yet 

Please link the PR with the PR of your comments, thanks

**Comment by rafaelbn** [comment]
Please rebase @sergio-incaser thanks

File: website_portal_sale/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    'data': [
+        'templates/website_portal_sale.xml',
+        'templates/website_portal.xml',
+        'templates/website.xml',
+        'security/ir.model.access.csv'
+    ],
+    'demo': [
+        'demo/sale_order.xml',
+    ],
+    'license': 'LGPL-3',
```
Comment (yajo) [inline website_portal_sale/__openerp__.py:32]: AFAIK, license is required in every file's header. Although you can use the reduced version available in the template. Not blocking anyway.

**Comment by yajo** [comment]
> [BUG-3] This module shows only the quotations and invoices created after installing the module and not all invoices of the res.partner. It doesn't shows the invoices created before installing it.

I think the problem is that you did not click on "Send Quotation" button. Draft quotations will never appear in portal, that's a good design.

Code & functional review, :+1:

**Comment by rafaelbn** [comment]
@Yajo 
- Login as admin:

```
+Go to Sales > Customer > More options > Portal Access Management and invite it
+to access at portal.
```
- After that go to user and copy the link for setting password


- As you can check user want to set his/her pass word but:

**Comment by yajo** [comment]
@rafaelbn That's fixed in https://github.com/OCA/website/pull/122. Uninstall that module and try again.

**Comment by yajo** [comment]
Please rebase to fix https://github.com/OCA/website/pull/111#issuecomment-160674487.

**Comment by yajo** [comment]
:+1:

**Comment by rafaelbn** [comment]
OK! Again tested in runbot, it looks :+1:

**Comment by rafaelbn** [comment]
@LeartS could you give us your last review please? Thanks

**Comment by JavierIniesta** [comment]
Tested in runbot :+1:

File: website_portal_sale/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    'name': 'Website Portal for Sales',
+    'category': 'Website',
+    'summary': (
+        'Add your sales document in the frontend portal (sales order'
+        ', quotations, invoices)'
+    ),
+    'version': '8.0.1.0.0',
+    'author': 'Odoo SA, '
+              'MONK Software, '
+              'Antiun Ingenieria SA, '
```
Comment (rafaelbn) [inline website_portal_sale/__openerp__.py:13]: Please @sergio-incaser make a PR to FIX this
s/Antiun Ingenieria SA,/Antiun Ingeniería S.L. (review template)
Thanks

**Pull request merged**